compiler/rustc_lint/src/types.rs

```diff
@@ -1117,6 +1117,19 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
                 FfiSafe
             }
 
+            ty::TyAlias(def_id, substs) => {
+                for ty in substs.types() {
+                    let ty = self.cx.tcx.normalize_erasing_regions(self.cx.param_env, ty);
+                    match self.check_type_for_ffi(cache, ty) {
+                        FfiSafe => {}
+                        r => return r,
+                    };
+                }
+                let ty = tcx.type_of(def_id);
+                let ty = self.cx.tcx.normalize_erasing_regions(self.cx.param_env, ty);
+                self.check_type_for_ffi(cache, ty)
```

Comment:
this can be simpler, i think

```suggestion
                let ty = tcx.bound_type_of(def_id).subst(substs);
                self.check_type_for_ffi(cache, self.cx.tcx.normalize_erasing_regions(self.cx.param_env, ty))
```